make_rpm

```diff
@@ -0,0 +1,142 @@
+usage="$(basename "$0") help [clean|cleanall]
+This script is used to create the Debian package of FoglAMP C++ '${plugin_name}' ${plugin_type} plugin
+
+Arguments:
+ help     - Display this help text
+ clean    - Remove all the old versions saved in format .XXXX
+ cleanall - Remove all the versions, including the last one"
+
+for i in "$@"
+do
+  case "$i" in
+      clean)
+          echo -n "Cleaning the build folder from older versions..."
+          find "${GIT_ROOT}/packages/build" -maxdepth 1 | grep '.*\.[0-9][0-9][0-9][0-9]' | xargs rm -rf
+          echo "Done."
+          exit 0
+          ;;
+      cleanall)
+          if [ -d "${GIT_ROOT}/packages/build" ]; then
+            echo -n "Cleaning the build folder..."
+            rm -rf ${GIT_ROOT}/packages/build/*
+            echo "Done."
+          else
+            echo "No build folder, skipping cleanall"
+          fi
+          exit 0
+          ;;
+      help)
+          echo "${usage}"
+          exit 0
+          ;;
+      *)
+        echo "Unrecognized option: $i"
+        exit 1
+        ;;
+  esac
+done
+
+# Get plugin version from VERSION
+version=`cat ${GIT_ROOT}/VERSION`
+foglamp_version=`cat ${GIT_ROOT}/foglamp.version | tr -d ' ' | grep 'foglamp_version' | head -1 | sed -e 's/\(.*\)version\(.*\)/\2/g'`
+BUILD_ROOT="${GIT_ROOT}/packages/build"
+rpm_release="1"
+
+arch=`arch`
+
+# Final package name
+pkg_name="foglamp-${plugin_type}-${plugin_name}"
+package_name="foglamp-${plugin_type}-${plugin_name}-${version}-${rpm_release}.${arch}"
+
+# Print the summary of findings
+echo "The package root directory is         : ${GIT_ROOT}"
+echo "The FogLAMP ${plugin_type} ${plugin_name} version is   : ${version}"
+echo "The package will be built in          : ${BUILD_ROOT}"
+echo "The package name is                   : ${package_name}"
+echo
+
+# Create the package directory. If a directory with the same name exists,
+# it is copied with a version number.
+
+# First, create the BUILD_ROOT folder, if necessary
+if [ ! -L "${BUILD_ROOT}" -a ! -d "${BUILD_ROOT}" ]; then
+    mkdir -p "${BUILD_ROOT}"
+fi
+
+cd "${BUILD_ROOT}"
+existing_pkgs=`find . -maxdepth 1 -name "${package_name}.????" | wc -l`
+existing_pkgs=$((existing_pkgs+1))
+new_stored_pkg=$(printf "${package_name}.%04d" "${existing_pkgs}")
+if [ -d "${package_name}" ]; then
+    echo "Saving the old working environment as ${new_stored_pkg}"
+    mv "${package_name}" "${new_stored_pkg}"
+fi
+mkdir "${package_name}"
+
+# Populate the package directory with RPM files
+# First with files common to all plugins
+echo -n "Populating the package and updating version file..."
+cd "${package_name}"
+cp -R ${GIT_ROOT}/packages/RPM/* .
+sed -i "s/__VERSION__/${version}/g" SPECS/foglamp.spec
+sed -i "s/__NAME__/${pkg_name}/g" SPECS/foglamp.spec
+sed -i "s/__ARCH__/${arch}/g" SPECS/foglamp.spec
+sed -i "s/Requires: foglamp/Requires: foglamp (${foglamp_version})/g" SPECS/foglamp.spec
```

Comment:
Requires sed is not correct, It may be fix like below

`sed -i "s/Requires:\([[:space:]]*\)foglamp/Requires:\1foglamp${foglamp_version}/g" SPECS/foglamp.spec`